File: app/src/main/java/edu/tjrac/swant/douyin/module/main/DouYinActivity.java
package edu.tjrac.swant.douyin.module.main;

import android.os.Bundle;
import android.support.v4.view.ViewPager;

import com.yckj.baselib.common.adapter.FragmentsPagerAdapter;

import butterknife.BindView;
import butterknife.ButterKnife;
import edu.tjrac.swant.unicorn.R;
import edu.tjrac.swant.unicorn.SharedStartActivity;

public class DouYinActivity extends SharedStartActivity {

    @BindView(R.id.vp) ViewPager mContent;

    FragmentsPagerAdapter adapter;

    CameraFragment camera;
    MainFragment main;
    UserInfoFragment userinfo;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_douyin_main);
        ButterKnife.bind(this);
        adapter = new FragmentsPagerAdapter(getSupportFragmentManager());
        camera = new CameraFragment();
        main = new MainFragment();
        userinfo = new UserInfoFragment();

        adapter.addFragment(camera, "camera");
        adapter.addFragment(main, "main");
        adapter.addFragment(userinfo, "userinfo");

        mContent.setAdapter(adapter);
        mContent.setCurrentItem(1);
        mContent.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

            }

            @Override
            public void onPageSelected(int position) {
                if (position == 0) {

                } else if (position == 1) {

                } else if (position == 2) {

                }
            }

            @Override
            public void onPageScrollStateChanged(int state) {

            }
        });
    }


}

File: app/src/main/java/edu/tjrac/swant/douyin/module/main/DouYinActivity.kt
package edu.tjrac.swant.douyin.module.main

import android.os.Bundle
import android.support.v4.view.ViewPager

import com.yckj.baselib.common.adapter.FragmentsPagerAdapter

import edu.tjrac.swant.unicorn.R
import edu.tjrac.swant.unicorn.SharedStartActivity
import kotlinx.android.synthetic.main.activity_douyin_main.*

class DouYinActivity : SharedStartActivity() {

    internal var adapter: FragmentsPagerAdapter?=null
    internal var camera: CameraFragment?=null
    internal var main: MainFragment?=null
    internal var userinfo: UserInfoFragment?=null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_douyin_main)
        adapter = FragmentsPagerAdapter(supportFragmentManager)
        camera = CameraFragment()
        main = MainFragment()
        userinfo = UserInfoFragment()

        adapter!!.addFragment(camera, "camera")
        adapter!!.addFragment(main, "main")
        adapter!!.addFragment(userinfo, "userinfo")

        vp.setAdapter(adapter)
        vp.setCurrentItem(1)
        vp.addOnPageChangeListener(object : ViewPager.OnPageChangeListener {
            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {

            }

            override fun onPageSelected(position: Int) {
                if (position == 0) {

                } else if (position == 1) {

                } else if (position == 2) {

                }
            }

            override fun onPageScrollStateChanged(state: Int) {

            }
        })
    }


}
